modules/solace/script.go

```diff
@@ -0,0 +1,100 @@
+package solace
+
+import (
+	"fmt"
+)
+
+// generateCLIScript generates a Solace CLI script for configuring VPN, users, queues, and topic subscriptions.
+// Reference: https://docs.solace.com/Admin-Ref/CLI-Reference/VMR_CLI_Commands.html
+func generateCLIScript(settings options) string {
+	if len(settings.queues) == 0 {
+		return ""
+	}
+
+	script := `enable
+configure
+`
+
+	// Create VPN if not default
+	if settings.vpn != defaultVpn {
+		script += generateVPNConfig(settings.vpn)
+	}
+
+	// Configure username and password
+	script += generateUserConfig(settings.username, settings.password, settings.vpn)
+
+	// Configure VPN Basic authentication
+	script += generateVPNAuth(settings.vpn)
+
+	// Configure queues and topic subscriptions
+	script += generateQueueConfig(settings.queues, settings.vpn)
+
+	return script
+}
+
+// generateVPNConfig creates the CLI commands for setting up a custom VPN
+func generateVPNConfig(vpn string) string {
+	return fmt.Sprintf(`create message-vpn %s
```

Comment:
question: would all of these be better done with a single template instead of multiple functions and internal logic?

I'm wondering if a WithScript might be the logical conclusion?

Reply:
If I understand you correctly, you suggest introducing a `WithScript` method where module users can provide their solace script themselves? 

That might be good flexibility indeed; however, I don't think the users would want to know/write the Solace script to just introduce a basic queue/subscription pair as it can be frustrating to get it working, so maintaining a well tested logic inside the module itself wouldn't hurt imho, that logic might also provide an example/starting point to other devs that want to contribute this solace module.

That being said, I'd be happy to hear more on that as myself had a hard time to get this script working for solace 😜

Reply:
Implemented the template approach, I _heavily inspired_ from the rabbitmq module where they used `embed` package. 
Let me know if that looks alright. 
https://github.com/testcontainers/testcontainers-go/blob/f5c13097653b3fbe4bdd6b980804bacc4fb20b88/modules/rabbitmq/rabbitmq.go#L204

fyi @mdelapenya